torchrl/envs/libs/gym.py

```diff
@@ -419,6 +419,16 @@ def _build_env(
             env = self._build_gym_env(env, pixels_only)
         return env
 
+    def read_action(self, action):
+        action = super().read_action(action)
+        if (
+            isinstance(self.action_spec, (OneHotDiscreteTensorSpec, DiscreteTensorSpec))
+            and action.size == 1
+        ):
+            # some envs require an integer for indexing
+            action = int(action)
```

Comment:
tiny: You may want to use action.item() as it is more efficient 
scalar = np.array([1])
timeit.timeit(lambda: scalar.item())
> 0.05105495895259082
timeit.timeit(lambda: int(scalar))
> 0.34526545903645456

Moreover, it seems this functionality will be deprecated in future:
> DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)